Dockerfile

```diff
@@ -69,56 +69,60 @@ ARG CABAL_VERSION=3.8.1.0
 RUN \
   set -o errexit -o xtrace; \
   if test -n "$CABAL_VERSION"; then \
-  ghcup install cabal "$CABAL_VERSION" --set; \
-  cabal --version; \
+    ghcup install cabal "$CABAL_VERSION" --set; \
+    cabal --version; \
   fi
 
-# Install Stack.
+# Configure Cabal.
 
-ARG STACK_VERSION=2.9.1
+ARG CABAL_STORE=/cabal-store
 RUN \
   set -o errexit -o xtrace; \
-  if test -n "$STACK_VERSION"; then \
-  ghcup install stack "$STACK_VERSION" --set; \
-  stack --version; \
+  if command -v cabal; then \
+    sudo mkdir --mode 0775 --parents --verbose "$CABAL_STORE"; \
+    sudo chown --verbose "$USER_NAME" "$CABAL_STORE"; \
+    sudo chgrp --verbose sudo "$CABAL_STORE"; \
+    cabal user-config init --augment "store-dir: $CABAL_STORE"; \
   fi
 
-# Install HLS.
+# Install Stack.
 
-ARG HLS_VERSION=1.8.0.0
+ARG STACK_VERSION=2.9.1
 RUN \
   set -o errexit -o xtrace; \
-  if test -n "$HLS_VERSION"; then \
-  ghcup install hls "$HLS_VERSION" --set; \
-  ghcup gc --hls-no-ghc; \
-  haskell-language-server-wrapper --version; \
+  if test -n "$STACK_VERSION"; then \
+    ghcup install stack "$STACK_VERSION" --set; \
+    stack --version; \
   fi
 
-# Configure Cabal.
+# Configure Stack.
 
-ARG CABAL_STORE=/cabal-store
+ARG STACK_ROOT=/stack-root
 RUN \
   set -o errexit -o xtrace; \
-  if command -v cabal; then \
-  sudo mkdir --mode 0775 --parents --verbose "$CABAL_STORE"; \
-  sudo chown --verbose "$USER_NAME" "$CABAL_STORE"; \
-  sudo chgrp --verbose sudo "$CABAL_STORE"; \
-  cabal user-config init --augment "store-dir: $CABAL_STORE"; \
+  if command -v stack; then \
+    sudo mkdir --mode 0775 --parents --verbose "$STACK_ROOT"; \
+    sudo chown --verbose "$USER_NAME" "$STACK_ROOT"; \
+    sudo chgrp --verbose sudo "$STACK_ROOT"; \
+    stack config set install-ghc --global false; \
+    stack config set system-ghc --global true; \
   fi
+ENV STACK_ROOT="$STACK_ROOT"
 
-# Configure Stack.
+# Install HLS.
 
-ARG STACK_ROOT=/stack-root
+ARG HLS_VERSION=1.8.0.0
 RUN \
   set -o errexit -o xtrace; \
-  if command -v stack; then \
-  sudo mkdir --mode 0775 --parents --verbose "$STACK_ROOT"; \
-  sudo chown --verbose "$USER_NAME" "$STACK_ROOT"; \
-  sudo chgrp --verbose sudo "$STACK_ROOT"; \
-  stack config set install-ghc --global false; \
-  stack config set system-ghc --global true; \
+  if test -n "$HLS_VERSION"; then \
+    if echo "$HLS_VERSION" | grep --extended-regexp --quiet '^[0-9a-f]{40}$'; then \
+      ghcup compile hls --jobs $(nproc) --cabal-update --ghc "$GHC_VERSION" --git-describe-version --git-ref "$HLS_VERSION" -- --ghc-options='+RTS -M1G -RTS'; \
```

Comment:
Can we use more than 1GB for the heap? The CodeBuild machine has roughly 16 GB of RAM.